### DISCUSSION TITLE
Any planning to supported duckdb?

### DISCUSSION THREAD

**Discussion by LittleKey** [sqlc-dev/sqlc#2515] [Feature Requests & Ideas] [answered]
Should sqlc add supported of [duckdb](https://github.com/duckdb/duckdb) ? That's awesome **embedded** RDBMS may instead of sqlite3 in OLAP scene.

The most stars go-driver for duckdb [go-duckdb](https://github.com/marcboeker/go-duckdb)

**Comment by kyleconroy** [comment, marked as answer]
We're big fans of duckdb, but don't have any current plans to integrate it into sqlc.

The go-duckdb package you linked is helpful, but we need access to the [internals](https://duckdb.org/internals/) to power parsing SQL and resolving types and references. Since DuckDB is open source, it's possible, but will be a significant amount of work.

**Reply by LittleKey** [reply]
Thanks for reply.

At least we know what the difficulties are and hope the integrate can be planned as soon as possible.

**Reply by casimcdaniels** [reply]
@kyleconroy

FYI: updated link is 

https://duckdb.org/docs/internals/overview.html

**Reply by zakaria-chahboun** [reply]
Hi guys any updates? Or roadmap?

**Reply by zfogg** [reply]
+1 please add support

**Comment by kyleconroy** [comment]
Support for `sqlc parse` and `sqlc analyze` landed in https://github.com/sqlc-dev/sqlc/pull/4585
